src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/PrimitiveWrapperAsUnknownObject.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Tests
+{
+    // Wrap a .net object as an UntypedObject.
+    // This will lazily marshal through the object as it's accessed.
+    [DebuggerDisplay("{_source}")]
+    public class PrimitiveWrapperAsUnknownObject : IUntypedObject
+    {
+        public readonly object _source;
+
+        public PrimitiveWrapperAsUnknownObject(object source)
+        {
+            _source = source;
+        }
+
+        public static UntypedObjectValue New(object source)
+        {
+            return FormulaValue.New(new PrimitiveWrapperAsUnknownObject(source));
+        }
+
+        public FormulaType Type
+        {
+            get
```

Comment:
what about boolean?

Reply:
I can add. This is existing test code (moved from elsewhere in the PR) .

Reply:
This is a good example why it  would be nice to have UntypedObject be more general - ie, we can't handle dates or other .net primitive types.